bundler/lib/bundler/settings.rb

```diff
@@ -329,16 +340,24 @@ def split_specific_setting_for(name)
     end
 
     def is_bool(name)
-      BOOL_KEYS.include?(name.to_s) || BOOL_KEYS.include?(parent_setting_for(name.to_s))
+      name = name.to_s
+      BOOL_KEYS.include?(name) || BOOL_KEYS.include?(parent_setting_for(name))
     end
 
     def is_string(name)
-      STRING_KEYS.include?(name.to_s) || name.to_s.start_with?("local.") || name.to_s.start_with?("mirror.") || name.to_s.start_with?("build.")
+      name = name.to_s
+      STRING_KEYS.include?(name) || name.start_with?("local.") || name.start_with?("mirror.") || name.start_with?("build.")
     end
 
     def to_bool(value)
       case value
-      when nil, /\A(false|f|no|n|0|)\z/i, false
+      when String
+        if value.match?(/\A(false|f|no|n|0|)\z/i)
+          false
+        else
+          true
+        end
```

Comment:
```suggestion
         value !~ /\A(false|f|no|n|0|)\z/i
```

I'm not sure if that's equally performant, but it saves a few lines.

Reply:
could also do `!value.match?(/\A(false|f|no|n|0|)\z/i)`

Reply:
I like your solution better. Docs say that match? doesn't update the globals, so it's probably slightly more performant.

Reply:
I did think about doing it on a single line. I know it's a few lines, but I find it a bit easier to read. could do as a ternary operator too

```ruby
value.match?(/\A(false|f|no|n|0|)\z/i) ? false : true
```

Checked w/ benchmark-ips and they are the same, so I'm open to whatever style y'all prefer.

Reply:
I also like @segiddins' version!

Reply:
I prefer one line, but don't care between ternary and `!` and sorry for starting the nitpick thread here. 😆